pkg/imgpkg/signature/fetch_signatures.go

```diff
@@ -17,36 +19,65 @@ type Finder interface {
 	Signature(reference name.Digest) (imageset.UnprocessedImageRef, error)
 }
 
+// NotFoundErr specific not found error
 type NotFoundErr struct{}
 
+// Error Not Found Error message
 func (s NotFoundErr) Error() string {
 	return "signature not found"
 }
 
+// Signatures Signature fetcher
 type Signatures struct {
 	signatureFinder Finder
 	concurrency     int
 }
 
+// NewSignatures constructs the Signature Fetcher
 func NewSignatures(finder Finder, concurrency int) *Signatures {
 	return &Signatures{
 		signatureFinder: finder,
 		concurrency:     concurrency,
 	}
 }
 
+// Fetch Retrieve the available signatures associated with the images provided
 func (s *Signatures) Fetch(images *imageset.UnprocessedImageRefs) (*imageset.UnprocessedImageRefs, error) {
 	signatures := imageset.NewUnprocessedImageRefs()
+	var imgs []lockconfig.ImageRef
+	for _, ref := range images.All() {
+		imgs = append(imgs, lockconfig.ImageRef{
+			Image: ref.DigestRef,
+		})
+	}
+	imagesRefs, err := s.FetchFromImageRef(imgs)
+	if err != nil {
+		return nil, err
+	}
+	for _, ref := range imagesRefs {
+		signatures.Add(imageset.UnprocessedImageRef{
+			DigestRef: ref.Image,
+			Tag:       ref.Annotations["tag"],
+		})
+	}
+
+	return signatures, err
+}
+
+// FetchFromImageRef Retrieve the available signatures associated with the images provided
+func (s *Signatures) FetchFromImageRef(images []lockconfig.ImageRef) (map[string]lockconfig.ImageRef, error) {
```

Comment:
FetchFromImageRef => FetchForImageRefs?

i would argue that we shouldnt be returning a map here. seems like an internal implementation. i dont see anyone consuming PrimaryLocation. (can still use a map to do uniquness check, but can return an array)

Reply:
done, I made it a slice all over